app/helpers/tag_helper.rb

```diff
@@ -0,0 +1,12 @@
+module TagHelper
+  def tag_colours
+    %w[primary info success olive warning orange danger pink purple violet]
+  end
+
+  def tag_colour(index)
+    # there are only 10 colours available for tags so we only care about
+    # index values between 0 and 9 when we access them.
+    colour_index = index.digits.first
```

Comment:
I would probably do a modulo here so that it cycles through, eg
```suggestion
    colour_index = index % tag_colours.length
```

it's effectively the last digit thanks to there being ten tags and the number being in base 10, but that's not guaranteed